### DISCUSSION TITLE
Command line support for cache statistics

### DISCUSSION THREAD

**Discussion by kediarajesh** [mortbopet/Ripes#257] [Ideas]
Hi,
I am using Ripes simulator for teaching computer architecture and found it quite helpful. It was good to see that the recent release (2.2.5) supports command line invocation and dumping the values of registers, etc. I was curious if we could also obtain cache statistics from the command line interface. This can help students to analyze various combinations of cache configurations and their effect on hit/miss rate in a better way than obtaining them manually through the GUI.

Regards,
Rajesh

**Comment by mortbopet** [comment]
>  I was curious if we could also obtain cache statistics from the command line interface.

Should definitely be possible and i see the use-case for it as well. I don't spend that much time actively developing Ripes any longer, so if this is a feature you'd like to see sooner rather than later, i'd suggest recommending e.g. a student to look into how the current execution stats are exported through the command line interface, and extend that to the cache statistics as well. It's probably not much more than an afternoons' worth of work.

**Reply by kediarajesh** [reply]
Thanks Morten. My course just got over last week, so urgency is not there!
If it comes as an easy change in any of your future releases, please consider it. Else, I will get it done through a student sometime.

Regards,
Rajesh
